src/platform/packages/shared/kbn-unified-tabs/src/components/tabs_bar_menu/tabs_bar_menu.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import {
+  EuiButtonIcon,
+  EuiPopover,
+  useGeneratedHtmlId,
+  EuiSelectableOption,
+  EuiSelectable,
+  EuiPopoverTitle,
+  EuiHorizontalRule,
+  EuiSelectableOptionsListProps,
+} from '@elastic/eui';
+import type { TabItem } from '../../types';
+import type { TabsBarProps } from '../tabs_bar';
+
+const getOpenedTabsList = (
+  tabItems: TabItem[],
+  selectedTab: TabItem | null
+): EuiSelectableOption[] => {
+  if (tabItems.length === 0) {
+    return [];
+  }
+
+  return tabItems.map((tab) => ({
+    label: tab.label,
+    checked: selectedTab && tab.id === selectedTab.id ? 'on' : undefined,
+    key: tab.id,
+  }));
+};
+
+const getRecentlyClosedTabsList = (tabItems: TabItem[]): EuiSelectableOption[] => {
+  if (tabItems.length === 0) {
+    return [];
+  }
+
+  return tabItems.map((tab) => ({
+    label: tab.label,
+    key: tab.id,
+  }));
+};
+
+interface TabsBarMenuProps {
+  onSelectOpenedTab: TabsBarProps['onSelect'];
+  selectedTab: TabsBarProps['selectedItem'];
+  openedTabs: TabsBarProps['items'];
+  recentlyClosedTabs: TabsBarProps['recentlyClosedItems'];
+}
+
+export const TabsBarMenu: React.FC<TabsBarMenuProps> = ({
```

Comment:
Looks like the tab preview blocks the clicks for top tabs menu items. Let's update the branch with main to see if https://github.com/elastic/kibana/pull/216016 fixes that.

![Mar-27-2025 16-18-39](https://github.com/user-attachments/assets/4fd595dc-2fc5-4b97-8afe-e0203a15a92f)

Reply:
Thanks for that catch! It must have actually be introduced by some main changes, because before merging mine it worked fine. I'll merge the freshest and check it

Reply:
Should work now!